### DISCUSSION TITLE
Notices in console about cookies

### DISCUSSION THREAD

**Discussion by cibulka** [vimeo/player.js#981] [Bug reports]
### Expected Behavior

Working player without notices in console. Also, could you explain in docs how the player uses cookies and what it actually tracks?

### Actual Behavior

This notice in Chrome's console:

```
56:1 A cookie associated with a cross-site resource at http://vimeo.com/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.
```

This notice appears also with setting `dnt: true` on init.

### Steps to Reproduce

https://nest-forgottenwar.cibulka.now.sh/cs/story/nuda-selfie-zelezo/120

Thanks!

**Comment by alexluong** [comment]
I have the same question too. Would love to be part of the discussion. Thank you!

**Comment by thewiremonkey** [comment]
Same question.

**Comment by ghost** [comment]
Seeing the same issue here

**Comment by mikkopori** [comment]
Same problem here. A solution would be good.

Perhaps adding a header like recommended here could be the solution?

https://stackoverflow.com/a/58320564/7225808

**Comment by teds31** [comment]
Is anyone working on this?

**Comment by luwes** [comment]
sorry for the late reply, we're aware of this warning in the console and are actively looking how this can be resolved. note that the warning is just informative, we have not experienced any issues with the player due to this new feature in Chrome.

**Comment by hhimanshu** [comment]
We are getting this warning too

**Comment by luwes** [comment]
@cibulka we made some changes that should resolve it, at least on the player.vimeo.com domain.
how is it looking on your end? thanks

https://player.vimeo.com/video/76979871

**Comment by joshmckibbin** [comment]
The issue still remains on player.vimeo.com domain

**Comment by luwes** [comment]
@joshmckibbin have you tried clearing browser cache? 
also try changing the volume of the player, this will reset the player cookie.

as a last resort clear all cookies under the .vimeo.com domain. load the player again and check.

**Comment by joshmckibbin** [comment]
@luwes actually it seems to be fixed now. Thanks!

**Comment by darylknight** [comment]
This still happens if you're embedding a background video from Vimeo Pro using the external download link. Header video on https://qeprize.org is causing it.

**Comment by ConversionWars** [comment]
Confirmed still happening for me with a vimeo video BG

**Comment by spidercreations** [comment]
Same issue when embedding Vimeo Pro videos into a background slide within Slider Revolution. https://brownstonepark.com and https://powderridgepark.com

**Comment by ryearsley** [comment]
Same issue. Using API to pull in videos and embed. https://www.breathehotyoga.com This will become a serious issue for our business especially since we're now depending more on video due to the Covid-19 measures in place. We are located in Seattle and all of our studios are shut down.

**Comment by Hanley1** [comment]
Google has started enforcing this in recent updates (https://www.chromium.org/updates/same-site) for some users. So now I'm hearing from customers who can't play our videos (example: https://www.audiblegenius.com/course/bb1/section_1/drums_1/earned). 

For these users, the associated console error is no longer a warning:

`A cookie associated with a cross-site resource at http://vimeo.com/ was set without the 'SameSite' attribute. It has been blocked, as Chrome now only delivers cookies with cross-site requests if they are set with 'SameSite=None' and 'Secure'. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.`

**Comment by nbuttarelli** [comment]
Any update in this issue? Any workaround? Thanks!

**Comment by afonseca08** [comment]
I'm seeing the same message with the embedded Vimeo Pro player, player.vimeo.com, any update on a fix?

**Comment by thedavidthomas** [comment]
Seeing same for both akamaized.net & vimeo.com. Using external mp4 links (Vimeo Pro) inside a HTML5 video tag for a background video

A cookie associated with a cross-site resource at http://vimeo.com/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.

A cookie associated with a cross-site resource at http://akamaized.net/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.

**Comment by squareclouds** [comment]
hi guys, any updates on this? same behaviour here... and im having strange problems on chrome as well, sometimes the videos dont load at all, and after many refreshes they run, but does not feel stable enough to go live like this. also it is hard to know if the issue is tied to the warning, or if it is in our code.

**Comment by martinweck** [comment]
+1 for me. Thanks for looking into it!
```
A cookie associated with a cross-site resource at http://vimeo.com/ was set without the `SameSite` attribute. A future release of Chrome will only deliver cookies with cross-site requests if they are set with `SameSite=None` and `Secure`. You can review cookies in developer tools under Application>Storage>Cookies and see more details at https://www.chromestatus.com/feature/5088147346030592 and https://www.chromestatus.com/feature/5633521622188032.
```

**Comment by Hanley1** [comment]
Starting to get more messages from visitors who can't load some of our videos. One of the visitors reported the same thing happening with Vimeo players on other sites as well. I think it's safe to say this has gone beyond just being a warning, and the reports will only increase over time as Chrome rolls out this update to more people.

**Comment by truskowc** [comment]
+1

“vuid” is the cookie, fyi

**Comment by danielucas** [comment]
+1 still no movement on this? 

Getting messages from users on our site as well that videos (using HTML5 player, not iframe embed) aren't playing.

**Comment by hemilsheth** [comment]
Any updates for Vimeo samesite cookie issue, I dont seem to find the solution, we have to urgently fix this as Google is going to roll out changes on July 14, 2020.

**Comment by luwes** [comment]
thanks for all the feedback, we are in process of resolving this issue. by the end of the day the `vuid` cookie will be updated to have the proper SameSite attribute.

however there are still several 3rd party cookies (Google Analytics) loaded when you have visited the main website vimeo.com, these are created under the cookie domain `.vimeo.com`. subsequently whenever a Vimeo player embed (player.vimeo.com) is loaded via the iframe those cookies are also loaded and Chrome will show the warning because they don't have the `SameSite` attribute yet. these 3rd party scripts will be updated by the vendors soon but it's worth to point out that these cookies are not used in the player and will not cause any issues.

please let me know if you have any questions.

**Comment by wesort** [comment]
Thanks. I know this has been a long time coming (long been seeing that
console warning) so it's good to see this getting resolved.

.ben.

**Comment by proton1k** [comment]
Just for your information, might be useful to someone coming now. We used 3-rd party service fo parsing and accepting cookies and it managed to parse Vimeo cookies into Marketing group. If a user rejected Marketing type cookies, the video was not loaded. Thus we fixed the issue by setting up the correct cookie type as Necessary so that unavoidably the video was never blocked.

**Comment by repher** [comment]
Hello! Is this resolved? A client just called and after a review, the problem still exists ...?
Thanks!
<img width="516" alt="Screenshot 2020-08-09 at 10 25 06" src="https://user-images.githubusercontent.com/2478563/89728092-e5605a00-da2a-11ea-8dc0-e610919d74e2.png">

**Comment by RaulRueda** [comment]
Over a year without a solution or response 🙄

**Comment by rkrishnan8594** [comment]
Hey all - apologies for the lack of activity on this issue. As @luwes mentioned in his last comment, these warnings are being caused by third party cookies set on vimeo.com, not cookies that are being set by the player, so this would not have an impact on playback. Regardless, the team has a ticket prioritized to look into adding the SameSite attribute to those cookies, which would resolve this warning.

**Comment by brown2020** [comment]
Has there been any progress on this? It has been 6 months.

**Comment by duncandejong** [comment]
<img width="766" alt="Schermafbeelding 2022-04-15 om 21 36 07" src="https://user-images.githubusercontent.com/3210606/163624444-ba76c920-267b-45ff-b1f0-aa2ad58e03dd.png">
Any updates?

**Comment by KjellMorgenstern** [comment]
I cleared all Vimeo cookies, obviously then player.js will not trigger the error.
I then went to vimeo.com, logged in and navigated along several vimeo.com/* pages.
Still no no cookie cross-site request / errors anymore on the site with embedded player.js.
This looks like the cookies on vimeo.com now have proper SameSite attributes that don't cause issues with player.js.

**Comment by sansari** [comment]
This is still happening. Still no solution?
